apps/src/templates/Dialog.jsx

```diff
@@ -138,8 +152,24 @@ const Dialog = React.createClass({
     onCancel: whenNoChildOfTypes(Buttons),
     confirmText: whenNoChildOfTypes(Buttons),
     onConfirm: whenNoChildOfTypes(Buttons),
+    confirmType: whenNoChildOfTypes(Buttons),
   }),
 
+  handleKeyDown(event) {
+    // Always focus the Cancel or Confirm button when tab is pressed, to prevent the
+    // user from selecting elements outside of the dialog.
+    if (event.key === 'Tab') {
+      const cancelButton = this.refs.buttons.getCancelButton();
+      const confirmButton = this.refs.buttons.getConfirmButton();
+      if ((document.activeElement === cancelButton)) {
+        confirmButton && confirmButton.focus();
+      } else {
+        cancelButton && cancelButton.focus();
+      }
+      event.preventDefault();
```

Comment:
I'm being paranoid and trying to make it so the first button you tab to is not the Confirm (aka Delete) button... but perhaps that's not necessary. Cancel will come first in our current implementation anyway, and having to press tab before enter is already somewhat of a hurdle to prevent accidentally deleting data, so I'll go with just tabbing through all the buttons as you had suggested.